fendermint/vm/interpreter/src/fvm/query.rs

```diff
@@ -99,3 +96,131 @@ where
         Ok((state, res))
     }
 }
+
+impl<DB> FvmMessageInterpreter<DB>
+where
+    DB: Blockstore + 'static + Send + Sync + Clone,
+{
+    /// Overestimation rate applied to gas to ensure that the
+    /// message goes through in the gas estimation.
+    const GAS_OVERESTIMATION_RATE: f64 = 1.25;
+    /// Default gas premium value. Inferred through a quick search through
+    /// InvokeEVM messages in filfox. The default value is only used if
+    /// the user hasn't specified a gas premium.
+    const DEFAULT_GAS_PREMIUM: u64 = 20000;
+    /// Gas search step increase used to find the optimal gas limit.
+    const GAS_SEARCH_STEP: f64 = 1.2;
+
+    fn estimate_gassed_msg(
+        &self,
+        state: &FvmQueryState<DB>,
+        msg: Box<Message>,
+    ) -> anyhow::Result<Message> {
+        let mut out = (*msg).clone();
```

Comment:
Why not take ownership of the message and return the estimate-filled value? 

I think it's a code smell to take a value by reference only to clone it immediately. You should take it by value and let the caller decide if they want to clone it, or give ownership over theirs to you.

Reply:
That makes total sense. Thank you for the feedback. I think this was a consequence of me not knowing Rust well. I thought I couldn't not use `Box` as raw reference. Will be fixed in the same commit.

Reply:
`Box` is basically just a thing you own that points at something on the heap, so that you can have a pointer instead of including the bytes in your data structures. I think you can pass `Box<T>` as a raw reference `&T` with `&` or `.as_ref()`, and also take ownership of its contents by moving them onto the stack, which consumes the `Box`.